service/storage_service.cc

```diff
@@ -2848,6 +2871,10 @@ future<> storage_service::join_token_ring(sharded<db::system_distributed_keyspac
         join_params.ignore_nodes = utils::split_comma_separated_list(_db.local().get_config().ignore_dead_nodes_for_replace());
     }
 
+    if (!_group0->joined_group0()) {
+        co_await check_join_params_on_joining_node(join_params);
```

Comment:
Can we check them on the recipient instead? If the node already exists in `none` state in the topology already features in the request and in the topology should match.

Reply:
Yes - in the previous impl where adding to group 0 was done only after finishing join_node_response it was necessary, but now I can do it solely on the join_node_request recipient.

Reply:
Removed, now verification happens on the join_node_request recipient.